File: got-collection/app/src/main/java/com/gotcollection/joaobb/gotcollection/api/service/CharacterService.java
package com.gotcollection.joaobb.gotcollection.api.service;

import com.gotcollection.joaobb.gotcollection.api.model.CharacterResultModel;
import com.gotcollection.joaobb.gotcollection.db.entity.CharacterEntity;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface CharacterService {

    @GET("characters")
    Call<CharacterEntity[]> getCharacters();

    @GET("characters/{name}")
    Call<CharacterResultModel> getCharacterByName(@Path("name") String name);
}

File: got-collection/app/src/main/java/com/gotcollection/joaobb/gotcollection/api/service/CharacterService.kt
package com.gotcollection.joaobb.gotcollection.api.service

import com.gotcollection.joaobb.gotcollection.api.model.CharacterResultModel
import com.gotcollection.joaobb.gotcollection.db.entity.CharacterEntity

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path

interface CharacterService {

    @GET("characters")
    fun getCharacters(): Call<Array<CharacterEntity>>

    @GET("characters/{name}")
    fun getCharacterByName(@Path("name") name: String): Call<CharacterResultModel>
}
